packages/testsuite/cypress/e2e/undertow/test-configuration-subsystem-undertow-ajp-listener.cy.ts

```diff
@@ -0,0 +1,66 @@
+describe("TESTS: Configuration => Subsystem => Undertow => Global settings", () => {
+  let managementEndpoint: string;
+
+  const testValues = {
+    serverName: "default-server",
+    ajpListener: "test-ajp",
+    ajpSocketBindings: "ajp",
+    allowedReqAttrsPatternExpression: "${NON-EXISTING:value.*}",
+    allowedReqAttrsPatternResolved: "value.*",
+  };
+  const address = ["subsystem", "undertow", "server", testValues.serverName, "ajp-listener", testValues.ajpListener];
+  const serverSelectors = {
+    serverListenersItem: "#undertow-server-listener-item",
+    ajpListenerItem: "#undertow-server-ajp-listener-item",
+  };
+  const ajpListenerPageSelectors = {
+    ajpListenersTableId: "undertow-server-ajp-listener-table",
+  };
+  const ajpListenerForm = {
+    id: "undertow-server-ajp-listener-form",
+    allowedReqAttrsPattern: "allowed-request-attributes-pattern",
+  };
+
+  before(() => {
+    cy.startWildflyContainer()
+      .then((result) => {
+        managementEndpoint = result as string;
+      })
+      .then(() => {
+        // create fixtures
+        cy.task("execute:cli", {
+          managementApi: managementEndpoint + "/management",
+          address: address,
+          operation: "add",
+          "socket-binding": testValues.ajpSocketBindings,
+        });
+      });
+  });
+
+  after(() => {
+    cy.task("stop:containers");
+  });
+
+  beforeEach(() => {
+    cy.navigateTo(managementEndpoint, "undertow-server;name=default-server");
+    // the form takes a brief moment to initialize
+    cy.wait(200);
+    cy.get(serverSelectors.serverListenersItem).click();
+    cy.get(serverSelectors.ajpListenerItem).click();
+  });
+
+  it("Test AJP Listener: allowed request attributes pattern", () => {
```

Comment:
Do test also negatived scenario? When is no possible to define AJP_ALLOWED_REQUEST_ATTRIBUTES_PATTERN 

What a user see and what happened when a preview mode is not allowed?

Reply:
If a preview mode is not allowed, the parameter is simply not there. However it should be visible all the time - once it is in `default` stability level.

I guess there is no negative scenario - it is a pattern. Only invalid value is invalid patter (i.e.  `(.*`), but this is parsed during service startup (after you click on the reload) and the WEB UI doesn't check it. Same for the CLI op - it fails to laod after the reload.

Reply:
It could be a negative scenarios when a user fill a wrong expression.

Reply:
I would expect some notification which say the expression is wrong.

Reply:
As i understand. You can configure a wrong expression without any warning then restart a server and then it will not start again. So you have a dead server which you can't reconfigure back because it is offline now. So you have to manually update the standalone.xml to revert you last change. 

I don't know if it is possible to check it in advance. I am just asking.

Reply:
We have discussed this offline. Reload produce an error - the service won't start because of wrong expression. Console is still up and you can edit the expression (CLI or Web console).